app/services/google_auth_service.rb

```diff
@@ -4,8 +4,12 @@ class GoogleAuthService < BaseAuthService
   class << self
     def find_user_by_token(google_id_token, create_user)
       super
+    rescue Google::Auth::IDTokens::SignatureError => e
+      raise FailedAuthentication, e.message
+    rescue Google::Auth::IDTokens::AudienceMismatchError => e
+      raise FailedAuthentication, e.message
     rescue Google::Auth::IDTokens::ExpiredTokenError => e
-      raise self::FailedAuthentication, e.message
+      raise FailedAuthentication, e.message
```

Comment:
If there's multiple exceptions that will have the same action, you can put them on the same line. Unless you did this for style purposes so the line's not too long?

```
rescue Google::Auth::IDTokens::SignatureError, Google::Auth::IDTokens::AudienceMismatchError, Google::Auth::IDTokens::ExpiredTokenError => e
  raise FailedAuthentication, e.message
end
```